### ISSUE TITLE
Multi user

### ISSUE THREAD

**Issue by danieldaeschle** [zadam/trilium#169]
It would be nice to have the function to create other users and share workspaces with them.
another nice function would be to share a note to public. everybody with this link can access this note.

**Comment by zadam** [comment]
This is a complicated topic. I don't want to do full multi-user because from my perspective it's a pandora's box which would pivot the project away from its focus on personal note taking.

But some limited way of sharing notes might be possible - e.g. you might be able to publish some subtree of notes to another user (read-only so no conflicts).

Another thing I was thinking about would be again publishing subtree of notes to public web (of course read only again) - that way you could manage simple website from Trilium. Both of these might leverage existing synchronization protocol.

But this is still pretty far I think

**Comment by zadam** [comment]
> another thing is: the look isn't that nice in my opinion. it would be so awesome if i can create own themes which are moveable into an folder or something. The look isn't modern. i know, i can change it with CSS in my note, but that isn't really a good solution in my opinion. just for small changes.

Yeah, I know look isn't the best since I don't really have required design skills. You're right that existing CSS change functionality is meant mainly for small tweaks but even full fledged theme would mean mostly rewriting CSS.

You can hack on the default theme by editing resources/app/src/views/index.ejs and resources/app/src/public/stylesheets/style.css. If there's some worthwhile custom theme we can then think of some way how to include it / support it in non-hacky way.

**Comment by danieldaeschle** [comment]
I can try to contribute. I have a little bit design skills ;)

**Comment by AndyCoach** [comment]
It's a year later.. is there any chance for a multi-user version of Trilium Notes? :)
My team of programmers would love it to pieces.

**Comment by zadam** [comment]
A year later we at least have FAQ entry: https://github.com/zadam/trilium/wiki/FAQ#multi-user-support

TL;DR: probably not going to happen

**Comment by AndyCoach** [comment]
Oops, sorry I missed the FAQ, my bad.
I will remain hopeful though.

**Comment by mschneidercloud** [comment]
I can perfectly understand your reasons to not implement Multi-User support.
But what about sort of an authentication proxy site?

As everyone can just spin up multiple trilium-instances using Docker (one for each user), it would be very convenient to have a common login-page. For that authentication-proxy one simply would have to configure a redirection-path, dependent of the entered username. The proxy-page then should simply:
1. _Redirect_ the user to his/her dedicated trilium-instance;
2. Pass the entered login-details (username and password) to that user's trilium-instance;

Authentication would still be handled by the individual trilium-instance...
...that way it would look as multi-user, without being multi-user by design.

**Comment by zadam** [comment]
@mschneidercloud what you describe does not sound very complex, but it does not sound very useful either. The real benefit of multi-user system is collaboration between users - ability to read someone else's notes, contribute to them etc, but that would be missing here.

I think "multi-tenancy" would be a better term for such system.

**Comment by martinkoutecky** [comment]
@zadam actually, multi-tenancy is not that useless :) I have a VPS which is already running low on RAM, and an extra trilium instance makes a difference (and I have one already running there). I know there are probably higher-priority items, but I just wanted to put it out there that it would be helpful if a single instance could allow multiple users to sign in and work on multiple documents, even without any collaboration.
